### DISCUSSION TITLE
Getting vim-matchquote + matchit (or matchup) working

### DISCUSSION THREAD

**Discussion by al-ce** [LazyVim/LazyVim#57] [Q&A] [answered]
Has anyone gotten [vim-matchquote](airblade/vim-matchquote) to play nice with a `%` jump-extension like `matchit` or [matchup](https://github.com/andymass/vim-matchup)? Whether I enable/disable `matchit` in LazyVim's `config/lazy.lua` disabled_plugins table, and however I install / configure `matchup`, I can't get both matchit/matchup and matchquote to work at once. For example, given this html:

```html
<div>"hello 'lazy' "</div>
```

in a Packer [minimal config](https://pastebin.com/YwUxCwau):
    - (`matchit` or `matchup`) + `matchquote` : `%` jumps between html tags AND `'` or `"` (matchquote expected behavior)

in a LazyVim starter config (+ [plugin module](https://pastebin.com/0KwvNQpW)):
    - `matchit` enabled: `%` jumps betwwen tags
    - (`matchit` or `matchup`) + `matchquote` : `%` jumps between `'` or `"` but NOT between tags

Also, I don't think this is related, but I need to specify a recent commit for matchup in its lazy.nvim table, or else it defaults to v7.0 (Dec 2021) and on a `CursorMove` event I get these errors from `noice` https://pastebin.com/raw/TjbJmFxA (the behavior described above doesn't change otherwise)

This isn't a huge hindrance, I just wonder if there's something I'm missing, like some options I have to enable that I'm taking for granted, etc. Thanks!

**Comment by folke** [comment, marked as answer]
I use vim-matchup myself and I don't see how that could work together with something like vim-matchquote.

Matchup always jumps to the nearest jumpable node. Both plugins set mappings for `%`.

For your other question, you can specify `version=false`, and maybe also urge the plugin owner to release a new version ;)

**Reply by al-ce** [reply]
Thanks for the suggestion about using the `version` option. Yes that makes sense about the conflicting mappings for matchup. Do you think there's something about the load order in the Packer based config that lets both matchquote and matchit work in different situations? Something like, 'if the cursor is under a `'` or `"`, map `%` to matchquote, else to matchit" that's not happening in a LazyVim based config?

This is the behavior I'm trying to replicate.

https://user-images.githubusercontent.com/23170004/212429380-1907523e-4a49-4a17-9c84-25b21ada2ed5.mov

**Reply by folke** [reply]
Looking at the matchquote code it does seem to fallback to existing matchit behavior, you need to make sure matchquote is loaded **after** matchup. https://github.com/airblade/vim-matchquote/blob/master/plugin/matchquote.vim

You're currently loading matchup on `BufReadPost` and matchquote with `lazy=false`, so that will load before matchup.

Maybe try setting both `lazy=false`

**Reply by al-ce** [reply]
Thank you! Setting both to false didn't do it but setting them in some way to make sure matchquote is loaded after matchup does give both functionalities. I should have thought of trying to mess with `event`. Right now I've got 

```lua
  {  "andymass/vim-matchup",  event = "BufReadPre" },
  {  "airblade/vim-matchquote", event = "BufReadPost" },
```

Really appreciate you taking the time to look at that, I don't expect you to answer every edge case and I was hoping some other user ran into this and could help me, but it's very nice of you to do so. LazyVim is really fun, I'm enjoying rebuilding my config from scratch with it. Cheers!

**Reply by folke** [reply]
Glad you got it working!

With `lazy=false`, you can additionally set a `piority=`. The higher the priority the earlier it would run.

Alternatively, you could also add matchup as a dependency to matchquote. That way, matchup would always be loaded before matchquote.

But your events defnitely also do the trick :)
